internal/argocd/health.go

```diff
@@ -240,28 +236,80 @@ func (h *applicationHealth) GetApplicationHealth(
 
 // stageHealthForAppSync returns the v1alpha1.HealthState for an Argo CD
 // Application based on its sync status.
-func stageHealthForAppSync(app *argocd.Application, revision string) (kargoapi.HealthState, error) {
-	switch {
-	case revision == "":
+func stageHealthForAppSync(
+	app *argocd.Application,
+	desiredRevisions []string,
+) (kargoapi.HealthState, error) {
+	var haveDesiredRevisions bool
+	for _, r := range desiredRevisions {
```

Comment:
Potential candidate for https://pkg.go.dev/slices#ContainsFunc

Reply:
🤦‍♂️ Fixed in a98204a103110fb3e4f25599af4f2f6de899ba31